rasterio/_env.pyx

```diff
@@ -68,11 +68,11 @@ cdef void log_error(CPLErr err_class, int err_no, const char* msg) with gil:
 # other platforms. Each calls log_error().
 IF UNAME_SYSNAME == "Windows":
     cdef void __stdcall logging_error_handler(CPLErr err_class, int err_no,
-                                              const char* msg):
+                                              const char* msg) with gil:
         log_error(err_class, err_no, msg)
 ELSE:
     cdef void logging_error_handler(CPLErr err_class, int err_no,
-                                    const char* msg):
+                                    const char* msg) with gil:
```

Comment:
I take that back. I'd forgotten that I haven't installed `rasterio` from source in Windows before, and probably not going to have a chance to figure that one out today.

I also should have said explicitly above: commenting out https://github.com/mapbox/rasterio/blob/master/rasterio/_env.pyx#L196 also didn't have any impact on my segfault, so I'm guessing there's a similar line elsewhere that's causing problems for me, and so was curious how you went about identifying the line causing problems on your system in case I might replicate any steps in debugging my (apparently separate) issue.